services/core/cards-publication/src/main/java/org/lfenergy/operatorfabric/cards/publication/services/CardProcessingService.java

```diff
@@ -246,6 +246,11 @@ boolean checkIsAllTimeSpanEndDateAfterStartDate(CardPublicationData c) {
         });
     }
 
+    public void deleteCards(Instant endDateBefore) {
+        List<CardPublicationData> cards = cardRepositoryService.findCardByEndDateBefore(endDateBefore);
+        this.deleteCards(cards);
```

Comment:
I guess it work well when have few cards but if we want to delete thousand of cards , we will parse and load in memory all the card , we shall find a way to delete in mongo via a query without having to load cards

Reply:
Done